### PR TITLE
[tests] Added smoke tests for conv solvers (ConvAsmBwdWrW1x1 and more). Some fixes.

### PR DESCRIPTION
**Pull request by atamazov**
@junliume https://github.com/ROCmSoftwarePlatform/MIOpen/labels/testing https://github.com/ROCmSoftwarePlatform/MIOpen/labels/urgency_high

Proposed reviewers: @averinevg @xinpin -- mostly to share knowledge, plus @shurale-nkn 

### TODO
- [x] Add `GFX110X_ENABLED` where necessary.

### This PR adds smoke tests for conv solvers:
- ConvAsmBwdWrW1x1 (with tuning)
- ConvAsmBwdWrW3x3 (with tuning)
- ConvAsmImplicitGemmGTCDynamicFwdXdlopsNHWC (with tuning)
- ConvAsmImplicitGemmGTCDynamicBwdXdlopsNHWC (with tuning)
- ConvAsmImplicitGemmGTCDynamicWrwXdlopsNHWC (with tuning)
- ConvAsmImplicitGemmGTCDynamicFwdDlopsNCHWC (with tuning)
- ConvCkIgemmFwdV6r1DlopsNchw (with tuning)
- ConvHipImplicitGemmV4R1Fwd (with tuning)
- ConvHipImplicitGemmV4R4Fwd (with tuning)
- ConvHipImplicitGemmV4R4WrW (with tuning)
- ConvHipImplicitGemmV4R1WrW (with tuning)
- ConvHipImplicitGemmBwdDataV4R1 (with tuning)
- ConvHipImplicitGemmBwdDataV1R1 (with tuning)
- ConvHipImplicitGemmBwdDataV4R1Xdlops (with tuning)
- ConvHipImplicitGemmForwardV4R4Xdlops (with tuning)
- ConvHipImplicitGemmForwardV4R4Xdlops_Padded_Gemm (with tuning)
- ConvHipImplicitGemmBwdDataV1R1Xdlops (with tuning)
- ConvHipImplicitGemmFwdXdlops (with tuning)
- ConvHipImplicitGemmForwardV4R5Xdlops (with tuning)
- ConvHipImplicitGemmWrwV4R4Xdlops (with tuning)
- ConvHipImplicitGemmWrwV4R4Xdlops_Padded_Gemm (with tuning)
- ConvMlirIgemmFwd/Bwd/WrW (with tuning). Removed test_conv_igemm_mlir_small.
- ConvMlirIgemmFwd/Bwd/WrWXdlops(with tuning). Removed test_conv_igemm_mlir_xdlops_small.
- ConvBinWinogradRxSf2x3 (with tuning)
- ConvBinWinogradRxSf2x3g1
- ConvBinWinogradRxSf3x2 (with tuning)

### By products:
- [CMake][Fix] Fix issue in set_var_to_condition, see https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1253#pullrequestreview-1218296540
- [NFC] ConvHipImplicitGemmBwdDataV1R1:
  - Improve W/A for issue #309.
  - Do not observe FP16_ALT attribute (it is irrelevant as FP16 is not supported).
- [tests][NFC]
  - Rename custom tests: smoke_conv_solver_Conv* -> smoke_solver_Conv*
  - Added SKIP_UNLESS_COMPOSABLEKERNEL option for custom tests
  - Added support for the "--output_type" option of test_conv.
  - Removed unnecessary MIOPEN_TEST_FLOAT_ARG from test_conv commands where TEST_CONV_VERBOSE_F/B/W is already used.
  - Rename couple of solver tests for clarity.
  - [NFC] Rename cmake variable: TEST_ANY_ERROR -> TEST_TUNING

### CODE REVIEW

**Comment by atamazov** [comment]
TODO: Add `GFX110X_ENABLED` when necessary.

**Comment by xinlipn** [comment]
@atamazov manually merge conflicting options in test/CMakeLists.txt, e.g. SKIP_UNLESS_COMPOSABLEKERNEL TEST_TUNING ?

**Comment by atamazov** [comment]
@xinlipn I will resolve conflicts soon. Please note that this PR is not ready for review anyway, because #1898 is not merged yet.

File: src/solver/conv_asm_1x1u_stride2.cpp [hunk 1]
```diff
@@ -99,6 +99,14 @@ inline static bool IsLinear(const int v)
     return L <= v && v <= H;
 }
 
+static inline size_t divide_round_plus_inf(const size_t x, const unsigned y)
+{
+    assert(x >= 0 && y > 0);
```
Comment (junliume) [inline src/solver/conv_asm_1x1u_stride2.cpp]: @atamazov cppcheck issue in the CI
```
[2023-01-11T01:22:45.356Z] /home/jenkins/workspace/MLLibs_MIOpen_PR-1906/src/solver/conv_asm_1x1u_stride2.cpp:104:14: warning: Unsigned expression 'x' can't be negative so it is unnecessary to test it. [unsignedPositive]
[2023-01-11T01:22:45.356Z]     assert(x >= 0 && y > 0);
[2023-01-11T01:22:45.356Z]              ^

Comment (atamazov) [reply src/solver/conv_asm_1x1u_stride2.cpp]: @junliume Thanks. Will fix when PR is ready for review.

**Comment by atamazov** [comment]
@junliume @johnny-keker Ready for CI testing & review/merge!

**Comment by junliume** [comment]
> @junliume @johnny-keker Ready for CI testing & review/merge!

Started CI on the latest commit.

**Comment by junliume** [comment]
> @junliume @johnny-keker Ready for the next round of CI testing.

@atamazov  this PR has some issues with `smoke_solver_ConvHipImplicitGemmFwdXdlops` on gfx908

[log.txt](https://github.com/ROCmSoftwarePlatform/MIOpen/files/10610065/log.txt)

**Comment by atamazov** [comment]
@junliume I see, thank you.

**Comment by atamazov** [comment]
@junliume 
> @atamazov this PR has some issues with `smoke_solver_ConvHipImplicitGemmFwdXdlops` on gfx908

The reason is #1971

**Comment by atamazov** [comment]
@junliume @johnny-keker Ready for the next round of CI testing.

**Comment by atamazov** [comment]
[Informative] I am working on resolving conflicts...

**Comment by atamazov** [comment]
@junliume @johnny-keker Ready for the next round of CI testing.

**Comment by junliume** [comment]
> @junliume @johnny-keker Ready for the next round of CI testing.

CI started

**Comment by atamazov** [comment]
@junliume @johnny-keker Ready for the next round of CI testing.

**Comment by atamazov** [comment]
@junliume @johnny-keker Wait... there is an issue in smoke_solver_ConvHipImplicitGemmV4R1

**Comment by junliume** [comment]
> @junliume @johnny-keker Wait... there is an issue in smoke_solver_ConvHipImplicitGemmV4R1

@atamazov  We recently updated it via https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2005

**Comment by atamazov** [comment]
@junliume Thanks, I see. This is different issue related to W/A for #936 in Jenkinsfile.

@junliume @johnny-keker Ready for the next round of CI testing.

**Comment by atamazov** [comment]
@junliume https://github.com/ROCmSoftwarePlatform/MIOpen/labels/TESTING_CI_PASSED can we go ahead and merge?

**Review by junliume — APPROVED** [review]

**Comment by atamazov** [comment]
@junliume Thanks!

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -1899,216 +1894,433 @@ if(MIOPEN_TEST_FLOAT)
...
+      MIOPEN_DEBUG_CONVOLUTION_ATTRIB_FP16_ALT_IMPL=0
+      MIOPEN_FIND_MODE=normal MIOPEN_DEBUG_FIND_ONLY_SOLVER=ConvBinWinogradRxSf2x3 $<TARGET_FILE:test_conv2d>
+      --input 1 40 20 20 --weights 20 20 3 3 --pads_strides_dilations 1 1 1 1 1 1 --group-count 2 ${MIOPEN_TEST_FLAGS_ARGS}
+)
+
+add_custom_test(smoke_solver_ConvBinWinogradRxSf2x3g1 GFX900_DISABLED GFX103X_ENABLED HALF_ENABLED SKIP_XNACK_ON
+    COMMAND MIOPEN_DEBUG_CONVOLUTION_ATTRIB_FP16_ALT_IMPL=0
+      MIOPEN_FIND_MODE=normal MIOPEN_DEBUG_FIND_ONLY_SOLVER=ConvBinWinogradRxSf2x3g1 $<TARGET_FILE:test_conv2d>
+      --input 1 40 20 20 --weights 20 40 3 3 --pads_strides_dilations 1 1 1 1 1 1 ${MIOPEN_TEST_FLAGS_ARGS}
+)
```
Comment (atamazov) [inline test/CMakeLists.txt:2291]: This is duplicate. Will be removed in next PR.

**Pull request merged**